array.c

```diff
@@ -651,12 +651,18 @@ ary_ensure_room_for_push(VALUE ary, long add_len)
 
 /*
  *  call-seq:
- *      ary.freeze -> ary
- *
- *  Calls Object#freeze on +ary+ to prevent any further
- *  modification. A RuntimeError will be raised if a modification
- *  attempt is made.
+ *    array.freeze -> self
  *
+ *  Freezes +self+; returns +self+:
+ *    a = []
+ *    a.frozen? # => false
+ *    a1 = a.freeze # => []
+ *    a.frozen? # => true
+ *    a1.equal?(a) # => true # Returned self
+ *
+ *  Raises an exception for an attempt to modify +self+:
```

Comment:
This maybe a little bit confusing since it implies `freeze` raises an exception, it doesn't. What raises an exception is the later attempt of modification. I understand the format of the other methods are like this, but IMO is not this case.

I'd suggest to rephrase as a note:

```
Note: Any further attempt of modifying an object frozen with freeze will raise an FrozenError exception
```

or something similar.

Reply:
Clarified, I think.